### ISSUE TITLE
lr_scheduler.get_last_lr before scheduler steps

### ISSUE THREAD

**Issue by stas00** [microsoft/DeepSpeed#782]
With dynamic scaling enabled the optimizer doesn't kick in until some 20th step, and so if `lr_scheduler.get_last_lr` gets called during that period it crashes with:

```
  File "/mnt/nvme1/code/huggingface/transformers-master/src/transformers/trainer.py", line 1134, in _maybe_log_save_evaluate
    self.lr_scheduler.get_last_lr()[0]
  File "/mnt/nvme1/code/github/00optimize/DeepSpeed-div-by-zero/deepspeed/runtime/lr_schedules.py", line 728, in get_last_lr
    assert getattr(self, '_last_lr', None) is not None, "need to call step() first"
```

While I can add a workaround it wouldn't be clean. I think this should be solved on the DeepSpeed level, since the user can't control that easily when the optimizer and thus scheduler starts stepping. 

I know I can force the optimizer to run right away with `"initial_scale_power": 1,` but it might not be what the user wants - and it's too error-prone to rely on a param config value.

Perhaps `get_last_lr` should return the initial lr value in such situations and a warning that this is what it did? 

Wrt workaround - is there a call I can make on the `engine` to see if optimizer has been run at least once already?

So ideally the scheduler should be aware of the optimizer state (since it is hidden behind the optimizer in the Deepspeed logic) and if optimizer was run, then all is good, if it wasn't - then return the initial lr.

I know I was the one who backported `get_last_lr` from pytorch, but back then I wasn't aware of this issue. So perhaps it needs to be reworked to handle this use case.

Thinking more - it should probably just return `0`, rather than initial lr - since this is what it actually is - as there was no stepping yet then LR==0. But it's uncertain whether this a clean design solution. What do you think?

And then later: this won't work either, because DeepSpeed needs to support pytorch Optimizers which will assert still.

So it sounds like try/except call in the Trainer is perhaps the only valid solution here, since it's the trainer that calls `lr_scheduler.get_last_lr`

Use case: https://github.com/huggingface/transformers/issues/10330#issuecomment-784468261

**Comment by stas00** [comment]
After thinking about it, my conclusion is that it's the trainer's responsibility to handle this exception, because we can't change pytorch optimizers.
